Add abstract database

Uncreated opened the pull request:
Добавление абстрактного класса коммуникации пользователей

Uncreated committed:
Немного правок стиля и добавление firebase

Uncreated committed:
Добавлен абстрактный класс коммуникации пользователей и сущности для него

vgonikhin reviewed (changes requested):
1. класс User - что такое keyId? Почему не просто id?
2. point - нечёткое название. какая точка?
3. не хватает как минимум поля mode и соответствующего enum {DRIVER,PASSENGER} чтобы понимать, в какой роли выступает юзер. И посмотри ещё мою диаграмму классов (на вкладке wiki) - может для текущей итерации ещё что-то понадобится
4. Раз уж мы с rx делаем, может void методы заменить на completable?
5. По первому коммиту - я бы гугловский json со всякими ключами не выкладывал. И если ключ google maps api ещё не вырезали, надо вырезать

Uncreated commented:
1. Да, на id заменил, не заливал коммит.
2. Изменил на CoordinatePoint.
3. Всё есть, классы Driver и Passenger
Остальное обсудим.

Uncreated committed:
Подправил имена классов, убрал google-services.json и внёс его в gitignore

Uncreated committed:
Замена типа методов с void на Completable в IUsersCommunication

Uncreated committed:
Изменена структура Communation модуля и связанных с ним сущностей

IUsersCommunication разбит на IDriverCommunication и IPassengerCommunication
Сущности запросов и ответов перенесены в другой пакет

Uncreated committed:
Добавлены реализации запроса и обновления позиций пользователей

Uncreated committed:
Правки полей и названий

Uncreated committed:
Добавлена реализация запросов пасажира

Uncreated committed:
merge add_abstract_database
